pallets/thea/src/session.rs

```diff
@@ -0,0 +1,57 @@
+use crate::{Config, Pallet};
+use frame_support::{log, traits::OneSessionHandler};
+use sp_core::{bounded::BoundedVec, Get};
+use sp_std::vec::Vec;
+
+impl<T: Config> sp_runtime::BoundToRuntimeAppPublic for Pallet<T> {
+	type Public = T::TheaId;
+}
+
+impl<T: Config> OneSessionHandler<T::AccountId> for Pallet<T> {
+	type Key = T::TheaId;
+
+	fn on_genesis_session<'a, I: 'a>(validators: I)
+	where
+		I: Iterator<Item = (&'a T::AccountId, T::TheaId)>,
+	{
+		let authorities = validators.map(|(_, k)| k).collect::<Vec<_>>();
+		// we panic here as runtime maintainers can simply reconfigure genesis and restart the
+		// chain easily
+		Self::initialize_authorities(&authorities).expect("Authorities vec too big");
+	}
+
+	fn on_new_session<'a, I: 'a>(_changed: bool, validators: I, queued_validators: I)
+	where
+		I: Iterator<Item = (&'a T::AccountId, T::TheaId)>,
+	{
+		// TODO: what happens if someone changes their session key
+		// and queued_validators != validators in next set.???
```

Comment:
Still remains?

Reply:
Pallet TODOs will be done in the next release